x/application/keeper/msg_server_delegate_to_gateway.go

```diff
@@ -36,6 +36,13 @@ func (k msgServer) DelegateToGateway(goCtx context.Context, msg *types.MsgDelega
 		return nil, sdkerrors.Wrapf(types.ErrAppGatewayNotFound, "gateway not found with address: %s", msg.GatewayAddress)
 	}
 
+	// Ensure the application is not already delegated to the maximum number of gateways
+	maxDelegatedParam := k.GetParams(ctx).MaxDelegatedGateways
+	if int64(len(app.DelegateeGatewayPubKeys)) >= maxDelegatedParam {
+		logger.Info("Application already delegated to maximum number of gateways: %d", maxDelegatedParam)
+		return nil, sdkerrors.Wrapf(types.ErrAppMaxDelegatedGateways, "application already delegated to maximum number of gateways: %d", maxDelegatedParam)
```

Comment:
1. I was under the impression (link missing - the need for the GH comment search tool returns :sweat_smile:) that the canonical way to import cosmossdk.io/errors was using the `errorsmod` alias.

2. The combination of this `Wrapf()` call and the base error results in quite a verbose error message:
    ```
    application already delegated to maximum number of gateways: 7: maximum number of delegated gateways reached
    ```
    Wdyt reword this one to something like?:

    ```suggestion
		return nil, errorsmod.Wrapf(types.ErrAppMaxDelegatedGateways, "application already delegated to %d gateways", maxDelegatedParam)
    ```
    Resulting in:
    ```
    application already delegated to 7 gateways: maximum number of delegated gateways reached
    ```

Reply:
This is errorsmod aliased to sdkerrors this was a decision olshansky and I had previously. He updated all occurances to be sdkerrors

Reply:
+1

I think we agreed to use sdkerrors instead of errorsmod from a comment @bryanchriswhite left on one of my PRs :)